radixdlt/src/integration/java/com/radixdlt/consensus/BFTTest.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * (C) Copyright 2020 Radix DLT Ltd
+ *
+ * Radix DLT Ltd licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.consensus;
+
+import com.radixdlt.consensus.checks.AllProposalsHaveDirectParentsCheck;
+import com.radixdlt.consensus.checks.LivenessCheck;
+import com.radixdlt.consensus.checks.NoSyncExceptionCheck;
+import com.radixdlt.consensus.checks.NoTimeoutCheck;
+import com.radixdlt.consensus.checks.SafetyCheck;
+import com.radixdlt.crypto.ECKeyPair;
+import com.radixdlt.middleware2.network.TestEventCoordinatorNetwork;
+import io.reactivex.rxjava3.core.Observable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class BFTTest {
+	private final int numNodes;
+	private final long time;
+	private final TimeUnit timeUnit;
+	private final int minNetworkLatency;
+	private final int maxNetworkLatency;
+	private final List<BFTCheck> checks = new ArrayList<>();
```

Comment:
I'm curious what your plans for dynamic configurations are, which may require different checks over time. I had a dynamic configuration in an earlier test where nodes randomly die over time, and this sort of dynamic is useful to test for. Will we be mapping configurations to sets of checks?

Reply:
haven't really gotten this far yet into thinking how that would look like

Reply:
Ah, so `CrashFaultNetworkTest` are staying as they are for now and not integrated with this new BFTTest interface (specifically, `given_7_correct_bfts_that_randomly_crash__then_correct_instances_should_make_progress_as_possible_over_1_minute`).

Reply:
yup